src/Model/Table/SentencesTable.php

```diff
@@ -438,21 +439,18 @@ public function findFilteredTranslations($query, $options) {
         });
     }
 
-    private function getextraTranslationsCount($sentence)
-    {
-        $maxDisplayed = 5;
-        $directTranslationCount = count($sentence->translations[0]);
-        $indirectTranslationCount = count($sentence->translations[1]);
-        return $directTranslationCount + $indirectTranslationCount - $maxDisplayed;
-    }
-
-    private function getExpandLabel($extraTranslationsCount)
+    public function getExpandLabel($total)
     {
-        return format(__n(
-            'Show 1 more translation',
-            'Show {number} more translations',
-            $extraTranslationsCount
-        ), array('number' => $extraTranslationsCount));
+        $extraTranslationsCount = $total - self::MAX_TRANSLATIONS_DISPLAYED;
+        if ($extraTranslationsCount > 0) {
+            return format(__n(
+                'Show 1 more translation',
+                'Show {number} more translations',
+                $extraTranslationsCount
+            ), array('number' => $extraTranslationsCount));
```

Comment:
I think it would be nice if that number is localized. 

While it is possible to use a helper (here `Cake\View\Helper\NumberHelper`) in a model, this demonstrates that `getExpandLabel` breaks the MVC philosophy where the model and the view shouldn't be coupled. Thus I don't think `getExpandLabel` should be a table method.

Reply:
Related bug: #2194.

Reply:
So this would be my solution if we were to localize the number:
https://github.com/Tatoeba/tatoeba2/tree/expand-label

I don't feel very happy about this code though but I don't see another way.

Reply:
There might be a way to get this done in Javascript using [angular-gettext](https://angular-gettext.rocketeer.be/), but that project is not maintained any more, and we’d have to find a way to implement NumberHelper in Javascript too. It feels like too much trouble and uncertainty for that one string. 

Maybe ecb2e3e323f20f4f099b96f838d96b93901b3b6b can be the basis of a solution. You don’t have to throw away all the the CakePHP serialization code, you can expand it. Look at `src/View/JsonView.php` in CakePHP’s code, it basically just runs `json_encode()`. So you could extend that class into, let’s say, `SentencesJsonView.php`, make it add the expand label before calling `json_encode()`, and then use `$this->RequestHandler->renderAs($this, 'sentences_json');` from the controller.

As for the views, you already have `SentencesHelper::sentenceForAngular()` that does a similar job, so you could have the expand label added from there too, and call that method from `SentencesJsonView.php`.

By the way, findFilteredTranslations has grown into something that does a lot more that filtering translations: it sets the expand label, as well as permissions and extra translations count. Maybe it’s a good opportunity to move all this out of the Sentence model too.